File: apps/api/app/schemas/match.py
"""Eşleştirme Ajanı endpoint şemaları"""
from typing import Optional
from pydantic import BaseModel, Field


class MatchRequest(BaseModel):
    listing_id: str = Field(..., description="Eşleştirilecek ilanın id'si")


class ScoreBreakdown(BaseModel):
    required: float
    nice_to_have: float
    seniority: float
    semantic_bonus: float


class MatchResponse(BaseModel):
    match_id: str
    listing_id: str
    score: float
    matched_skills: list[str]
    missing_skills: list[str]
    score_breakdown: Optional[ScoreBreakdown] = None
    cached: bool

File: apps/web/types/match.ts
export interface MatchRequest {
  listing_id: string;
}

export interface ScoreBreakdown {
  required: number;
  nice_to_have: number;
  seniority: number;
  semantic_bonus?: number;
}

export interface MatchResponse {
  match_id: string;
  listing_id: string;
  score: number;
  matched_skills: string[];
  missing_skills: string[];
  score_breakdown: ScoreBreakdown | null;
  cached: boolean;
}
